smart-contracts/smart-contract/tests/main-contract_test.ts

```diff
@@ -48,6 +51,134 @@ const err_not_proposed_notifier = '(err u124)';
 const err_already_notifier = '(err u125)';
 const err_no_voting_period = '(err u129)';
 
+Clarinet.test({
+  name: 'Get All Data Waiting 300 Miners',
+  async fn(chain: Chain, accounts: Map<string, Account>) {
+    const deployer = accounts.get('deployer')!;
+    let waiting_list = [];
+    let block = chain.mineBlock([]);
+    assertEquals(block.receipts.length, 0);
+    assertEquals(block.height, 2);
+
+    for (let i = 1; i <= 299; i++) {
+      const miner = accounts.get(`wallet_${i}`)!;
+      block = chain.mineBlock([
+        Tx.contractCall(CONTRACT_NAME, ASK_TO_JOIN, [types.ascii(miner.address)], miner.address),
+      ]);
+      if (i == 1) waiting_list.push(`${miner.address}`);
+      else waiting_list.push(` ${miner.address}`);
+    }
+    block = chain.mineBlock([Tx.contractCall(CONTRACT_NAME, GET_WAITING_LIST, [], deployer.address)]);
+    assertEquals(block.receipts[0].result, `[${waiting_list}]`);
+    assertEquals(block.receipts.length, 1);
+    assertEquals(block.height, 302);
+
+    for (let i = 1; i <= 299; i++) {
+      const miner = accounts.get(`wallet_${i}`)!;
+      block = chain.mineBlock([
+        Tx.contractCall(CONTRACT_NAME, VOTE_POSITIVE_JOIN, [types.principal(miner.address)], deployer.address),
+        Tx.contractCall(CONTRACT_NAME, TRY_ENTER_POOL, [], miner.address),
+      ]);
+    }
+
+    block = chain.mineBlock([Tx.contractCall(CONTRACT_NAME, GET_PENDING_LIST, [], deployer.address)]);
```

Comment:
@BowTiedJohn can we add another assertEquals to check the length of the pending list variable?

Reply:
We don't keep the pending list locally, but I am doing the checks for it through waiting list (since everyone in waiting list is being transferred to pending). Also, on the next line, we check that the pending list is equal to our local waiting list, which makes the requested asserts a little bit useless. I will add them though, as an extra layer of safety.